src/rdfTypes/GeometryInfoHelpersImpl.h

```diff
@@ -284,29 +313,7 @@ struct MetricLengthVisitor {
   CPP_template(typename T)(
       requires ad_utility::SimilarTo<T, AnyGeometry<CoordType>>) double
   operator()(const T& geom) const {
-    using enum AnyGeometryMember;
-    // `AnyGeometry` is a class from `pb_util`. It does not operate on an enum,
-    // this is why we use our own enum here. The correct matching of the integer
-    // identifiers for the geometry types with this enum is tested in
-    // `GeometryInfoTest.cpp`.
-    switch (AnyGeometryMember{geom.getType()}) {
-      case POINT:
-        return MetricLengthVisitor{}(geom.getPoint());
-      case LINE:
-        return MetricLengthVisitor{}(geom.getLine());
-      case POLYGON:
-        return MetricLengthVisitor{}(geom.getPolygon());
-      case MULTILINE:
-        return MetricLengthVisitor{}(geom.getMultiLine());
-      case MULTIPOLYGON:
-        return MetricLengthVisitor{}(geom.getMultiPolygon());
-      case COLLECTION:
-        return MetricLengthVisitor{}(geom.getCollection());
-      case MULTIPOINT:
-        return MetricLengthVisitor{}(geom.getMultiPoint());
-      default:
-        AD_FAIL();
-    }
+    return std::visit(MetricLengthVisitor{}, anyGeometryToParsedWkt(geom));
```

Comment:
That now really is a deep copy which we do not want.... I know why i wanted the `visit` function:)